ui/components/app/signature-request-siwe/format-message-params.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * Takes in a parsed Sign-In with Ethereum Message (EIP-4361)
+ * and generates an array of label-value pairs
+ *
+ * @param {Object} parsedMessage - A parsed SIWE message with message contents
+ * @param {Function} t - i18n function
+ * @returns {Array} An array of label-value pairs with the type of the value as the label
+ */
+export const formatMessageParams = (parsedMessage, t) => {
```

Comment:
[REQ] This file throws me off just a tad because I expect mostly react components in the components folder. I do believe we have some dated examples of having helper method files inside component folders but we should try and prevent extending this pattern. I think it would be helpful to create a SIWE file in shared/modules at the root of the repository that contains this method, and potentially the contents of scripts/lib/siwe/detecteSIWE.js. I know in the case of the later its only used in the backend but I think bundling these things together into one SIWE module file would be beneficial

Reply:
This was originally in [`ui/helpers/utils/format-message-params.js`](https://github.com/MetaMask/metamask-extension/pull/14438#discussion_r881110964). I could update this to exist in `shared/modules`

Reply:
Moved SIWE modules

Reply:
I like the new file created in `shared/modules` to hold all SIWE utils. My issue is that this `formatMessageParams` logic is specific to one component file and does not look like it will be reused. I think this logic can be moved directly into the component that uses this. 

This is a non-blocker

cc: @brad-decker @skgbafa